src/reboost/hpge/hit.py

```diff
@@ -0,0 +1,463 @@
+
+    Returns
+    -------
+    a new column for the hit table either :class:`Array`, :class:`ArrayOfEqualSizedArrays` or :class:`VectorOfVectors`.
+
+    Note
+    ----
+    In future the passing of local variables (pars,hpge,reg) to the evaluation should be make more generic.
+    """
+    if local_dict is None:
+        local_dict = {}
+
+    if info["mode"] == "eval":
+        # replace hit.
+        expr = info["expression"].replace("hit.", "")
+
+        msg = f"evaluating table with command {expr} and local_dict {local_dict.keys()}"
+        log.debug(msg)
+
+        col = table.eval(expr, local_dict)
+
+    elif info["mode"] == "function":
+        proc_func, globs = utils.get_function_string(info["expression"])
+
+        # add hit table to locals
+        local_dict = {"hit": table} | local_dict
+
+        msg = f"evaluating table with command {info['expression']} and local_dict {local_dict.keys()} and global dict {globs.keys()}"
+        log.debug(msg)
+        col = eval(proc_func, globs, local_dict)
+
+    else:
+        msg = "mode is not recognised."
+        raise ValueError(msg)
+    return col
+
+
+def build_hit(
+    file_out: str,
+    list_file_in: list,
+    out_field: str,
+    in_field: str,
+    proc_config: dict,
+    pars: dict,
+    n_evtid: int | None = None,
+    start_evtid: int = 0,
+    buffer: int = 1000000,
+    gdml: str | None = None,
+    metadata_path: str | None = None,
+    merge_input_files: bool = True,
+    has_global_evtid: bool = False,
+) -> None:
+    """
+    Read incrementally the files compute something and then write output
+
+    Parameters
+    ----------
+        file_out
+            output file path
+        list_file_in
+            list of input files
+        out_field
+            lh5 group name for output
+        in_field
+            lh5 group name for input
+        proc_config
+            the configuration file for the processing. Must contain the fields `channels`, `outputs`, `step_group` and operations`.
+            Optionally can also contain the `locals` field to extract other non-JSON serializable objects used by the processors.
+            For example:
+
+            .. code-block:: json
+
+               {
+                    "channels": [
+                        "det000",
+                        "det001",
+                        "det002",
+                        "det003"
+                    ],
+                    "outputs": [
+                        "t0",
+                        "truth_energy_sum",
+                        "smeared_energy_sum",
+                        "evtid"
+                    ],
+                    "step_group": {
+                        "description": "group steps by time and evtid.",
+                        "expression": "reboost.hpge.processors.group_by_time(stp,window=10)"
+                    },
+                    "locals": {
+                        "hpge": "reboost.hpge.utils(meta_path=meta,pars=pars,detector=detector)"
```

Comment:
```suggestion
                        "hpge": "reboost.hpge.utils.get_hpge(meta_path=meta,pars=pars,detector=detector)"
```
isn't there a function name missing?

Reply:
yeah it was a mistake, i will add